egs/mls_english/ASR/zipformer/decode.py

```
@@ -0,0 +1,1085 @@
+            decoding_graph.scores *= params.ngram_lm_scale
+        else:
+            word_table = None
+            decoding_graph = k2.trivial_graph(params.vocab_size - 1, device=device)
+    else:
+        decoding_graph = None
+        word_table = None
+
+    if "modified_beam_search" in params.decoding_method:
+        if os.path.exists(params.context_file):
+            contexts = []
+            for line in open(params.context_file).readlines():
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)